PWGHF/TableProducer/candidateCreator3Prong.cxx

```diff
@@ -342,7 +345,25 @@ struct HfCandidateCreator3ProngExpressions {
       if (flag == 0) {
         // Printf("Checking Ds± → K± K∓ π±");
         if (RecoDecay::isMatchedMCGen(particlesMC, particle, pdg::Code::kDS, array{+kKPlus, -kKPlus, +kPiPlus}, true, &sign, 2)) {
-          flag = sign * (1 << DecayType::DsToKKPi);
+          if (decaychannelDs == 2) {
+            flag = sign * (1 << DecayType::DsToKKPi);
+          } else {
+            Int_t pdgDau = 0;
+            if (decaychannelDs == 0) {
+              pdgDau = 333;
+            } else if (decaychannelDs == 1) {
+              pdgDau = 313;
+            }
+            std::vector<int> arrAllDaughtersIndex;
+            RecoDecay::getDaughters(particle, &arrAllDaughtersIndex, array{333, 211}, 1);
+            for (auto indexDaughterI : arrAllDaughtersIndex) {
+              auto candidateDaughterI = particlesMC.rawIteratorAt(indexDaughterI - particlesMC.offset()); // daughter particle
+              auto PDGCandidateDaughterI = candidateDaughterI.pdgCode();
+              if (PDGCandidateDaughterI == pdgDau) {
+                flag = sign * (1 << DecayType::DsToKKPi);
+              }
+            }
+          }
```

Comment:
A similar thing should be done also at the reconstructed MC level

Reply:
Hi @fcatalan92, we didn't do that for the recostructed MC level because we had the same result (with flag 0,1,2) utilizing the cut for the Phi in the dpl-configuration file. So we didn't think that it could be necessary, of course we can do that also there.

Reply:
Yes, please. We should not rely on the selection using the Phi invariant mass (which could be set to different values) to tag the decay in the MC. The fact the result is correct is just by chance.